File: 01-Class-Content/21-regionalized-content/Java/01-Activities/08-Stu_Classes-and-Objects/Solved/console/Student.java
package console;

import java.util.HashMap;

class Student {

  /* Make these package private for ease of use-this way, we don't have to
   *   define getters. */
  String firstName;

  String lastName;

  String middleName;

  String address;

  String email;

  String phoneNumber;

  Student (HashMap<String, String> studentProxy) {
    this.firstName = studentProxy.getOrDefault("first name", "N/A");

    this.lastName = studentProxy.getOrDefault("last name", "N/A");

    this.middleName = studentProxy.getOrDefault("middle name", "N/A");

    this.address = studentProxy.getOrDefault("address", "N/A");

    this.email = studentProxy.getOrDefault("email", "N/A");

    this.phoneNumber = studentProxy.getOrDefault("phone number", "N/A");
  }

  Student (String firstName, String lastName, String middleName,
          String address, String email, String phoneNumber) {

    this.firstName = firstName;

    this.lastName = lastName;

    this.middleName = middleName;

    this.address = address;

    this.email = email;

    this.phoneNumber = phoneNumber;

  }

  void report() {
    System.out.println(
        String.format("The student's name is %s %s %s.", 
          this.firstName, this.middleName, this.lastName));

    System.out.println(
        String.format("The student's address is %s.", this.address));

    System.out.println(
        String.format("The student's email is %s.", this.email));

    System.out.println(
        String.format("The student's phone number is %s.", this.phoneNumber));
  }

}

File: 01-Class-Content/21-regionalized-content/C#/01-Activities/04-Classes-and-Objects/Solved/Student.cs
using System;

namespace Students
{
  public class Student
  {
    public string firstName { get; set; }

    public string lastName {get; set; }

    public string middleName {get; set; }

    public string email { get; set; }

    public string address { get; set; }

    public string phoneNumber { get; set; }

    public static int created { get; private set; } = 0;

    public Student ()
    {
      Student.created += 1;
    }

    public void PrintAttributes ()
    {
      Console.WriteLine("This student's first name is " + this.firstName + ".");
      Console.WriteLine("This student's last name is " + this.lastName + ".");
      Console.WriteLine("This student's middle name is " + this.middleName + ".");

      Console.WriteLine("This student's email is " + this.email + ".");
      Console.WriteLine("This student's address is " + this.address + ".");
      Console.WriteLine("This student's phoneNumber is " + this.phoneNumber + ".");
    }

  }

}
